src/test/java/org/opensearch/integTest/ToolIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+                    + "\",\n"
+                    + "    \"description\": \"test model\",\n"
+                    + "    \"connector_id\": \""
+                    + connectorId
+                    + "\"\n"
+                    + "}"
+            );
+        Response response = executeRequest(request);
+
+        String resp = readResponse(response);
+
+        return JsonParser.parseString(resp).getAsJsonObject().get("model_id").getAsString();
+    }
+
+    private String setupConversationalAgent(String modelId) throws IOException {
+        Request request = new Request("POST", "/_plugins/_ml/agents/_register");
+        request
+            .setJsonEntity(
+                "{\n"
+                    + "  \"name\": \"integTest-agent\",\n"
+                    + "  \"type\": \"conversational\",\n"
+                    + "  \"description\": \"this is a test agent\",\n"
+                    + "  \"llm\": {\n"
+                    + "    \"model_id\": \""
+                    + modelId
+                    + "\",\n"
+                    + "    \"parameters\": {\n"
+                    + "      \"max_iteration\": \"5\",\n"
+                    + "      \"stop_when_no_tool_found\": \"true\",\n"
+                    + "      \"response_filter\": \"$.completion\"\n"
+                    + "    }\n"
+                    + "  },\n"
+                    + "  \"tools\": [\n"
+                    + "    {\n"
+                    + "      \"type\": \""
+                    + toolType()
+                    + "\",\n"
+                    + "      \"name\": \""
+                    + toolType()
+                    + "\",\n"
+                    + "      \"include_output_in_agent_response\": true,\n"
+                    + "      \"description\": \"tool description\"\n"
+                    + "    }\n"
+                    + "  ],\n"
+                    + "  \"memory\": {\n"
+                    + "    \"type\": \"conversation_index\"\n"
+                    + "  }\n"
+                    + "}"
+            );
+        Response response = executeRequest(request);
+
+        String resp = readResponse(response);
+
+        return JsonParser.parseString(resp).getAsJsonObject().get("agent_id").getAsString();
+    }
+
+    public static Response executeRequest(Request request) throws IOException {
+        RequestOptions.Builder builder = RequestOptions.DEFAULT.toBuilder();
+        builder.addHeader("Content-Type", "application/json");
+        request.setOptions(builder);
+        return client().performRequest(request);
+    }
+
+    public static String readResponse(Response response) throws IOException {
+        StringBuilder sb = new StringBuilder();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(response.getEntity().getContent()))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+        }
+        return sb.toString();
```

Comment:
Is it possible to use third party method like `IOUtils.toString()` to simplify this code block?

Reply:
done, update to OpenSearch built-in class instead `org.opensearch.common.io.Streams.readAllLines(ins)`